### ISSUE TITLE
RNN is not overfitting on simple data

### ISSUE THREAD

**Issue by SSS135** [fchollet/keras#5237]
I am trying to predict class for each number in input vector. There are 3 classes. Class 1, if input value changed from 0 to 1. Class 2, if it changed from 1 to 0. Class 0 otherwise.

After second epoch and onward accuracy is stuck at 0.8824. Higher number of training epoch does not change anything. I have tried switching `LSTM` to `GRU` or `SimpleRNN`, this changes nothing.

I've spend more than a day trying to make it work. What could be a problem? 
Thanks!

```python
# complete code for training
from keras.models import Sequential
from keras.layers import Dense, LSTM, TimeDistributed
from keras.utils.np_utils import to_categorical
import numpy as np

np.random.seed(1337)

X = np.array([0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0])
Y = np.array([0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0])
Y_cat = to_categorical(Y, 3).reshape((1, len(X), 3))
X_r = X.reshape((1, len(X), 1))

model = Sequential()
model.add(LSTM(32, input_dim=1, return_sequences=True))
model.add(LSTM(32, return_sequences=True))
model.add(LSTM(32, return_sequences=True))
model.add(TimeDistributed(Dense(3, activation='softmax')))

model.compile(optimizer='rmsprop',
              loss='categorical_crossentropy',
              metrics=['accuracy'])

model.fit(X_r, Y_cat, nb_epoch=10)

model.predict_classes(X_r) # will print array filled with zeros
```

**Comment by SSS135** [comment]
Answered [here](https://stackoverflow.com/questions/41977498/rnn-is-not-overfitting-on-simple-data).
